src/utils/alignment-D3-support.js

```diff
@@ -17,104 +29,71 @@ export function getTickPosition(tick) {
-}
-
-export function getTickValueMinutes(tick) {
-  const value = tick.querySelector('text').innerHTML;
-  return parseInt(value.split(':')[0], 10) +
-    (parseInt(value.split(':')[1], 10) / 60);
-}
-
-export function getShapePositionRect(shape) {
-  // the x, y attributes for each rect are from the top-left of the shape.
-  // compute the mid-value for a coordinate to compare to axis tick
-  let half, x, y;
-
-  half = parseFloat(shape.getAttribute('width')) / 2;
-  x = parseFloat(shape.getAttribute('x')) + half;
-
-  half = parseFloat(shape.getAttribute('height')) / 2;
-  y = parseFloat(shape.getAttribute('y')) + half;
-
-  return { x: x, y: y};
+export function getTickValue(item, dataType) {
+  let val = item.querySelector('text').innerHTML;
+  switch (dataType) {
+    case null:
+      break;
+    case 'minute':
+      val = parseInt(val.split(':')[0], 10) +
+        (parseInt(val.split(':')[1], 10) / 60);
+      break;
+    case 'month':
+      val = months.indexOf(val.toLowerCase());
+      break;
+    case 'thousand':
+      val = val.split(',').join('');
+      break;
+    default:
+      break;
+  }
+  return parseFloat(val);
 }
 
-export function getShapePositionRectBar(shape) {
-  // the x, y attributes for each rect are from the top-left of the shape.
-  // compute the mid-value for a coordinate to compare to x-axis tick
-
-  // TODO: rects are computed at y + 6 because
-  // the fCC Bar Chart appears to have misalignment.
-  let x, y, width;
-
-  width = parseFloat(shape.getAttribute('width'));
-  x = parseFloat(shape.getAttribute('x')) + (width / 2);
-  // fCC Bar Chart pen is at most 6px off on the y-axis
-  y = parseFloat(shape.getAttribute('y')) + 5.5;
-
-  return { x: x, y: y, width: width };
+export function getShapeValue(item, attribute, dataType) {
+  let val;
+  switch (dataType) {
+    case null:
+      val = item.getAttribute(attribute);
+      break;
+    case 'year':
+      val = new Date(item.getAttribute(attribute)).getFullYear();
+      break;
+    case 'minute':
+      val = new Date(item.getAttribute(attribute)).getMinutes() +
+        (new Date(item.getAttribute(attribute)).getSeconds() / 60);
+      break;
+    case 'month':
+      val = ( isNaN(parseInt(item.getAttribute(attribute), 10)) ?
+        months.indexOf(item.getAttribute(attribute).toLowerCase()) :
+        item.getAttribute(attribute)
+      );
+      break;
+    default:
+      val = item.getAttribute(attribute);
+  }
+  return parseFloat(val);
 }
 
-export function getShapePositionCircle(shape) {
-  let x, y;
-
-  x = parseFloat(shape.getAttribute('cx'));
-
-  y = parseFloat(shape.getAttribute('cy'));
-
-  return { x: x, y: y};
+export function getShapePosition(item, dimension, positionType) {
+  let half, pos = parseFloat(item.getAttribute('' + dimension + ''));
+  switch (positionType) {
+    case 'topLeft':
+      // bar
+      half = 0;
+      break;
+    case 'center':
+      let attr = !dimension.match(/x/g) ? 'height' : 'width';
```

Comment:
What's this line doing? I guess it's trying to match with anything but the letter 'x' and if that's the case `attr` gets assigned 'height'? Or 'width' if it does match? Also, why are we using a logical not (!) instead of `dimension.match(/x/g) ? 'width' : 'height'`? I'm sure there is a reason, it's just not obvious from the code.